apps/web/components/I18nLanguageHandler.tsx

```diff
@@ -7,6 +7,8 @@ import { trpc } from "@calcom/trpc/react";
 export function useViewerI18n() {
   return trpc.viewer.public.i18n.useQuery(undefined, {
     staleTime: Infinity,
+    refetchOnWindowFocus: false,
+    refetchOnMount: false,
```

Comment:
in react-query v4 i think it behaves differently. here's a reproduction of refetching while window focus, same behavior can be seen while switching routes also

[Screencast 2023-04-27 00:03:07.webm](https://user-images.githubusercontent.com/84864519/234670808-86fa1a4c-52a8-4eec-8183-9f55b7738974.webm)

Reply:
I think that's because app-providers.tsx wasn't using userVieweri18n which has staleTime set to infinite.
I tested your branch and removing the two props still fixes the refetching thing,

Reply:
nice catch @hariombalhara 😍 , updated the pr removing additional properties